### ISSUE TITLE
WCAG 2.2 Violation - There are labels present in the interface that are not programmatically associated with their corresponding controls.

### ISSUE THREAD

**Issue by ivrouwe-ubc** [ubc/wp-comments-chat-ui#4]
### Issue

There are labels present in the interface that are not programmatically associated with their corresponding controls. As a result, the controls are missing accessible names (despite having visible names adjacent to them).

#### This violates the following WCAG (Web Content Accessibility Guidelines) 2.2 Success Criteria:

- [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG22/#name-role-value)

### Instances

1. #38 
2. #39 

### User Impact

Some users - such as screen reader users or braille display users - will be unable to determine the names of these controls and what their purpose is.

#### Further information:

1. [The "Benefits" section of "Understanding Success Criterion 4.1.2: Name, Role, Value | WAI | W3C"](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html#benefits)

#### Using the Chrome developer tools:

1. Inspect the control using Chrome's developer tools.
2. Activate the "Accessibility" tab (for the element) within the developer tools.
3. Under the "Name" portion of the "Computed Properties" section, notice that "From label", "aria-labelledby", and "aria-label" are all listed as "Not specified" (and "Labeled by" is also not set - if the control is a form field). This indicates that a robust accessible name (that works for all assistive technologies) has not been set on the control.

### How To Fix The Issue

Provide the control with a descriptive accessible name using one of the following methods:

1. (If the form is an `<input>`, `<textarea>`, or `<select>` element):
    1. Ensure that the control has a unique `id` attribute set on it.
    2. Ensure that there is a descriptive `<label>` element for the control included alongside the control.
    3. Set the `<label>` element's `for` attribute value to match the value of the control's `id` value.
2. (If the control is a `<button>` element): set the `<button>` element's text content to be a descriptive label for the control, ensuring that the `aria-labelledby` and `aria-label` attributes are not present on the `<button>`.
3. For any other cases:
    1. Ensure that the labelling element has a unique `id` attribute value set on it.
    2. Add `aria-labelledby` to the control, setting its value to match the value of the labelling element's `id` attribute. If the `aria-label` attribute is present on the control: remove it.

#### WCAG 2.2 Conformant Code Example

Not available/not applicable.
